products/jbrowse-web/src/tests/BasicLinearGenomeView.test.js

```diff
@@ -259,6 +292,46 @@ describe('valid file tests', () => {
     expect(centerLineInfo.offset).toEqual(120)
   })
 
+  it('bookmarks current region', async () => {
+    const pluginManager = getPluginManager()
+    const state = pluginManager.rootModel
+    const { findByTestId, findByText } = render(
+      <JBrowse pluginManager={pluginManager} />,
+    )
+
+    const viewMenu = await findByTestId('view_menu_icon')
+    fireEvent.click(viewMenu)
+    fireEvent.click(await findByText('Bookmark current region'))
+    expect(state.session.bookmarkedRegions[0].start).toEqual(100)
+    expect(state.session.bookmarkedRegions[0].end).toEqual(140)
+  })
+
+  it('navigates to bookmarked region from widget', async () => {
+    const pluginManager = getPluginManager()
+    const state = pluginManager.rootModel
+    state.session.addBookmark({
+      start: 200,
+      end: 240,
+      refName: 'ctgA',
+      assemblyName: 'volvox',
+    })
+    const { findByTestId, findByText } = render(
+      <JBrowse pluginManager={pluginManager} />,
+    )
+    // need this to complete before we can try to navigate
+    await findByTestId('htsTrackEntry-volvox_alignments')
+
+    const viewMenu = await findByTestId('view_menu_icon')
+    fireEvent.click(viewMenu)
+    fireEvent.click(await findByText('Open bookmark widget'))
+    fireEvent.click(await findByText('ctgA:200..240'))
+    const newRegion = state.session.views[0].getSelectedRegions(
+      state.session.views[0].leftOffset,
+      state.session.views[0].leftOffset,
```

Comment:
Just curiosity, was this meant to be rightOffset? is this getting just one bp?

Reply:
Ahh, yes! I ended up checking the block key was valid, but yeah that should be `rightOffset`. Will fix!

Reply:
cool! looks good :) 👍